### PR TITLE
[Feature] Add MultiHumanData

### PR DESCRIPTION
**Pull request by ttxskk**
This PR mainly focuses on adding a new data structure for multi-human training.

- [x] Add a new data structure for multi-human training
- [x] Update unitest 
- [x] Add the Doc for MultiHumanData
- [x] Updata Doc for HumanData

### CODE REVIEW

File: docs/human_data.md [hunk 1]
```diff
@@ -204,3 +204,41 @@ dict_of_tensor = human_data.to()
...
+
+Based on HumanData, MultiHumanData adds a new key named `'optional'` as follows:
+
+```python
+'optional': {
+        'type': dict,
+        'slice_key': 'frame_range',
+        'dim': 0
+    }
+```
```
Comment (LazyBusyYang) [inline docs/human_data.md:220]: `frame_range` is more like a key and `optional` is more like an attribute.

File: mmhuman3d/data/data_converters/coco.py [hunk 2]
```diff
@@ -18,7 +24,11 @@ class CocoConverter(BaseConverter):
     <https://arxiv.org/abs/1405.0312>`__ .
     """
 
-    def convert(self, dataset_path: str, out_path: str) -> dict:
+    def convert(self,
+                dataset_path: str,
+                out_path: str,
+                multi_human_data: bool = False,
```
Comment (LazyBusyYang) [inline mmhuman3d/data/data_converters/coco.py:30]: Bool type var `multi_human_data` has a lower name of class `MultiHumanData`, it looks like an instance of  class `MultiHumanData`. I suggest something like `use_multi_human_data` or ` enable_multi_human_data`.

File: mmhuman3d/data/data_structures/multi_human_data.py [hunk 3]
```diff
@@ -0,0 +1,478 @@
+import logging
+import pickle
+from enum import Enum
+from typing import Any, TypeVar, Union
+
+import numpy as np
+from mmcv.utils import print_log
+
+from mmhuman3d.data.data_structures.human_data import HumanData
```
Comment (LazyBusyYang) [inline mmhuman3d/data/data_structures/multi_human_data.py:9]:
```suggestion
from mmhuman3d.data.data_structures.human_data import HumanData, _HumanData, _KeyCheck
```

File: mmhuman3d/data/data_structures/multi_human_data.py [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+        """New an instance of HumanData.
+
+        Args:
+            cls (HumanData): HumanData class.
+
+        Returns:
+            HumanData: An instance of Hu
+        """
+        ret_human_data = super().__new__(cls, args, kwargs)
+        # setattr(ret_human_data, '__data_len__', -1)
```
Comment (LazyBusyYang) [inline mmhuman3d/data/data_structures/multi_human_data.py:48]: Useless sentences can be removed.

File: mmhuman3d/data/data_structures/multi_human_data.py [hunk 5]
```diff
@@ -0,0 +1,478 @@
...
+                    value = np.hstack([value, np.ones([value.shape[0], 1])])
+                    tmp_data_dict[key] = value
+                else:
+                    tmp_data_dict[key] = value
+            self.update(tmp_data_dict)
+            self.__set_default_values__()
+
+    @property
+    def instance_num(self) -> int:
+        """Get the human instance num of this MultiHumanData instance. In
```
Comment (LazyBusyYang) [inline mmhuman3d/data/data_structures/multi_human_data.py:183]:
```suggestion
        """Get how many human are there in this MultiHumanData instance. In
```
The word `instance` is too frequent.

File: mmhuman3d/data/data_converters/coco.py (hunk 2 above)
Comment (ttxskk) [reply mmhuman3d/data/data_converters/coco.py]: done

File: mmhuman3d/data/data_structures/multi_human_data.py (hunk 3 above)
Comment (ttxskk) [reply mmhuman3d/data/data_structures/multi_human_data.py]: done

File: mmhuman3d/data/data_structures/multi_human_data.py (hunk 4 above)
Comment (ttxskk) [reply mmhuman3d/data/data_structures/multi_human_data.py]: done

File: mmhuman3d/data/data_structures/multi_human_data.py (hunk 5 above)
Comment (ttxskk) [reply mmhuman3d/data/data_structures/multi_human_data.py]: done

File: docs/human_data.md (hunk 1 above)
Comment (ttxskk) [reply docs/human_data.md:220]: remove `optional` and add a new key named `'frame_range'`
```python
'frame_range': {
        'type': np.ndarray,
        'shape': (-1, 2),
        'dim': 0
}
```

**Review by LazyBusyYang — APPROVED** [review]

**Pull request merged**